store/views.py

```diff
@@ -1,40 +1,47 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect
 from django.shortcuts import get_object_or_404
 from store.models import *
 from django.contrib.auth.decorators import login_required
 from django.http import JsonResponse
 from django.views.decorators.csrf import csrf_exempt
+from datetime import date
 
 # Create your views here.
 
+
 def index(request):
     return render(request, 'store/index.html')
 
+
 def bookDetailView(request, bid):
     template_name = 'store/book_detail.html'
     context = {
-        'book': None, # set this to an instance of the required book
-        'num_available': None, # set this to the number of copies of the book available, or 0 if the book isn't available
+        'book': None,  # set this to an instance of the required book
+        # set this to the number of copies of the book available, or 0 if the book isn't available
+        'num_available': None,
     }
     # START YOUR CODE HERE
-    
-    
+    context['book'] = Book.objects.get(pk=bid)
```

Comment:
This may fail with invalid book ID given in POST request, and would lead to server error. Expected behavior is to inform user with Not found (404) error.